### PR TITLE
Switch from MultiThreadedDebug to MultiThreadedDebugDll

### PR DESCRIPTION
**Pull request by Alan-Jowett**
This is a prerequisite for: #287 

Signed-off-by: Alan Jowett <alanjo@microsoft.com>

### CODE REVIEW

File: docs/GettingStarted.md [hunk 1]
```diff
@@ -64,6 +64,12 @@ a kernel debugger (KD) attached and running, or test signing is enabled. (It is
 releases of eBPF for Windows will eventually be production signed at some point in the future after
 security hardening is completed.)
 
+Ensure that the matching versions of the Microsoft C Runtime are installed on the machine. Versions of the Microsoft C Runtime are available on the developer machine, located in %ProgramFiles(x86)%\Microsoft Visual Studio\2019\Enterprise\VC\Redist\MSVC\version, where version is updated with each patch of Visual Studio.
+
+With version 14.29.30133, the full path is:
+1) Release - ```%ProgramFiles(x86)%\Microsoft Visual Studio\2019\Enterprise\VC\Redist\MSVC\14.29.30133\vc_redist.x64.exe```
+2) Debug - ```%ProgramFiles(x86)%\Microsoft Visual Studio\2019\Enterprise\VC\Redist\MSVC\14.29.30133\debug_nonredist\x64\Microsoft.VC142.DebugCRT```
```
Comment (dthaler) [inline docs/GettingStarted.md:71]: No such file on my machine.  Can we point to an online URL to install from?
Maybe https://support.microsoft.com/en-us/topic/the-latest-supported-visual-c-downloads-2647da03-1eea-4433-9aff-95f26a218cc0 ?

File: docs/SelfHostedRunnerSetup.md [hunk 2]
```diff
@@ -14,13 +14,14 @@ Self-hosted runners are necessary as GitHub-hosted runners don't have the requis
...
    1) ```./config.cmd --url https://github.com/microsoft/ebpf-for-windows --token <action runner token>```
 8) Change action runner service to run as "LocalSystem".
    1) Open services.msc.
    2) Locate "GitHub Action Runner ...".
    3) Right Click -> Properties -> Log On.
    4) Change "Log on as:" to "Local System Account".
-8) Install the [Visual C++ Runtime Files](https://docs.microsoft.com/en-us/visualstudio/releases/2019/redistribution#visual-c-runtime-files)
-9) Reboot the runner.
+9)  Install the [Visual C++ Runtime Files](https://docs.microsoft.com/en-us/visualstudio/releases/2019/redistribution#visual-c-runtime-files)
+10)  Install the debug version of the Visual C++ Runtime files from ```"%ProgramFiles(x86)%\Microsoft Visual Studio\2019\Enterprise\VC\Redist\MSVC\14.29.30133\debug_nonredist\x64\Microsoft.VC142.DebugCRT"``` on a machine with Visual Studio installed.
```
Comment (dthaler) [inline docs/SelfHostedRunnerSetup.md:26]: No such file at that path on my machine

File: docs/vm-setup.md [hunk 3]
```diff
@@ -26,6 +26,13 @@
     2. Do `bcdedit.exe -set TESTSIGNING ON`.
     3. Restart the VM so that the change will be applied.
 
+4. Install the matching version of the Microsoft Visual C++ Runtime:
+Versions of the Microsoft C Runtime are available on the developer machine, located in %ProgramFiles(x86)%\Microsoft Visual Studio\2019\Enterprise\VC\Redist\MSVC\version, where version is updated with each patch of Visual Studio.
```
Comment (dthaler) [inline docs/vm-setup.md:30]: No such path on my machine.  Specifically we do NOT require Enterprise edition.  Community edition should work.

**Review by shankarseal — APPROVED** [review]

File: docs/GettingStarted.md (hunk 1 above)
Comment (Alan-Jowett) [reply docs/GettingStarted.md]: Build script now copies the VC Runtime to the build output folder.

**Review by dthaler — APPROVED** [review]

**Pull request merged**